src/oop/object_interactions/medium/q4/QuestionText.md

```diff
@@ -0,0 +1,53 @@
+# oop-object_interactions-medium-q4
+
+- Java question
+
+## Question Text
+
+You are working as a vet assistant in an animal hospital, and sometimes you encounter an animal that requires a blood
+transfusion in order to save its life. Suppose you were given the following test class. Make a class called **Snail**
+with attributes **bloodType** and **volume** that are initialized by the constructor. This class has a method called
+**transfusionFrom()** that checks if the two animals have the same blood type and whether the donor has enough blood
+before starting the transfusion. If both conditions pass, then 1L of blood is transferred from the donor to the
+recipient.
+
+## Snail Class
+
+The `Snail` class currently has the following items:
+
+- Attributes:
+    - A `char` called **bloodType**
+    - A `double` called **volume**
+    - None of these attributes should be accessible from other classes.
+
+- Constructors:
+    - A constructor that take a `char` and a `double` to initialize the **bloodType** and **volume** attributes.
+    - It is accessible from other classes.
+
+- Methods:
+    - Getter ans Setter methods for **bloodType** and **volume**, and  **toString** method that prints the object
+      information.
+    - These methods are accessible from other classes.
+    -
+
+You must add the following items to the class:
+
+- Methods:
+    - A `transfusionFrom` method that take another **Snail** object to do a blood tansfer.
+    - It should check if the blood types are a match and if not print `Blood type incompatible`.
+    - If the types are compatible it should say `Blood transfusion starting...`
```

Comment:
it feels a bit odd to have this message and then a message saying the transfusion cannot happen, maybe switch around when these statements get printed

Reply:
I think the purpose here was to make students write nested if else statements which it lose that if that changes.

Reply:
If we just change the contents of the message then that element is still retained. Change it to something like "Checking if donor has enough blood for a transfusion."